packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -773,20 +819,23 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with AutomaticK
                 : CupertinoTheme.of(context).brightness == Brightness.light
                     ? _kDisabledBackground
                     : CupertinoColors.darkBackgroundGray,
-            child: TextSelectionGestureDetector(
-              onTapDown: _handleTapDown,
-              onForcePressStart: _handleForcePressStarted,
-              onForcePressEnd: _handleForcePressEnded,
-              onSingleTapUp: _handleSingleTapUp,
-              onSingleLongTapStart: _handleSingleLongTapStart,
-              onSingleLongTapMoveUpdate: _handleSingleLongTapMoveUpdate,
-              onSingleLongTapEnd: _handleSingleLongTapEnd,
-              onDoubleTapDown: _handleDoubleTapDown,
-              onDragSelectionStart: _handleMouseDragSelectionStart,
-              onDragSelectionUpdate: _handleMouseDragSelectionUpdate,
-              onDragSelectionEnd: _handleMouseDragSelectionEnd,
-              behavior: HitTestBehavior.translucent,
-              child: _addTextDependentAttachments(paddedEditable, textStyle, placeholderStyle),
+            child: Listener(
+              onPointerDown: _updateLastUsedDeviceKind,
```

Comment:
It seems like to @dkwingsmt and me that a better way to do this would be to add the pointer kind to the details for the gesture detector callbacks, and look at the device kind when needed.  Then you wouldn't need the extra listener, or to maintain a cache of the device kind.

Reply:
I gave him the opposite advice previously :D

https://github.com/flutter/flutter/pull/29683#issuecomment-475003862

because I wasn't sure where you guys wanted to go with altering signatures everywhere. If you guys were going there anyway, then what you're suggesting makes sense.

Reply:
Oh, sorry, I didn't notice that.  Yeah, we're going to have to go there anyhow, since there are going to be other cases of needing to distinguish.  The good thing is that it won't alter a lot of the signatures if there are details objects already, since we can just add it to the details that are already provided.

Reply:
Sounds reasonable. Sorry for the back and forth @mdebbar.

Reply:
@gspencergoog thanks for chiming in. @xster it's all good :)

I'll have the necessary changes ready for review by EOD.